internal/server/server.go

```diff
@@ -37,11 +37,17 @@ type Config struct {
 	NodePrefix      netip.Prefix
 	ServerAddress   netip.Addr
 	ServerEndpoint  netip.AddrPort
+	PeerProvider    PeerProvider
 	JoinToken       string
 	ServerPublicKey wgtypes.Key
 	Logger          *zap.Logger
 }
 
+// PeerProvider provides wireguard over GRPC peer addresses.
+type PeerProvider interface {
```

Comment:
this shouldn't be done this way, we need to use `PeerEvent`s to configure peers

this is the integration point with external peer handling, one more layer of indirection doesn't help us, but it will be more confusing

Reply:
Removed. Now we pass the virtual ip in the new `VirtualAddr` field inside `PeerEvent` and handle this in `Device.handlePeerEvent`.